cmake/cplex_interface/CMakeLists.txt

```diff
@@ -32,4 +70,5 @@ TARGET_LINK_LIBRARIES(${PROJECT_NAME}
     cplex
     concert
     dqrobotics
+    dl
```

Comment:
Can you remind me what this `dl` is for?

Reply:
@mmmarinho I am not sure. However, that library is required to compile. Otherwise, I have the following errors:

```shell
18:21:43: Running steps for project cplex_interface_test...
18:21:43: Starting: "/home/juanjqo/Qt/Tools/CMake/bin/cmake" --build /home/juanjqo/Documents/My_repositories/cpp-examples/cmake/build-cplex_interface-Desktop_Qt_6_3_1_GCC_64bit-Debug --target all
[1/1 3.5/sec] Linking CXX executable cplex_interface_test
FAILED: cplex_interface_test 
: && /home/linuxbrew/.linuxbrew/bin/g++-11 -g -rdynamic CMakeFiles/cplex_interface_test.dir/cplex_interface_test.cpp.o -o cplex_interface_test -L/opt/ibm/ILOG/CPLEX_Studio221/cplex/lib/x86-64_linux/static_pic   -L/opt/ibm/ILOG/CPLEX_Studio221/concert/lib/x86-64_linux/static_pic   -L/opt/ibm/ILOG/CPLEX_Studio221/bin/glnxa64 -Wl,-rpath,/opt/ibm/ILOG/CPLEX_Studio221/cplex/lib/x86-64_linux/static_pic:/opt/ibm/ILOG/CPLEX_Studio221/concert/lib/x86-64_linux/static_pic:/opt/ibm/ILOG/CPLEX_Studio221/bin/glnxa64  -lilocplex  -lcplex  -lconcert  -ldqrobotics  -lpthread && :
/home/linuxbrew/.linuxbrew/bin/ld: /opt/ibm/ILOG/CPLEX_Studio221/cplex/lib/x86-64_linux/static_pic/libcplex.a(load_dll_static_patched.o): in function `_Init_MKL_Loader':
load_dll.c:(.text+0x319f): undefined reference to `dladdr'
/home/linuxbrew/.linuxbrew/bin/ld: /opt/ibm/ILOG/CPLEX_Studio221/cplex/lib/x86-64_linux/static_pic/libcplex.a(mkl_memory_patched.o): in function `mkl_serv_set_memory_limit':
mkl_memory.c:(.text+0x5d1): undefined reference to `dlopen'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x5ee): undefined reference to `dlsym'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x62f): undefined reference to `dlclose'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x642): undefined reference to `dlsym'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x658): undefined reference to `dlsym'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x746): undefined reference to `dlerror'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x7b0): undefined reference to `dlopen'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x864): undefined reference to `dlopen'
/home/linuxbrew/.linuxbrew/bin/ld: /opt/ibm/ILOG/CPLEX_Studio221/cplex/lib/x86-64_linux/static_pic/libcplex.a(mkl_memory_patched.o): in function `mkl_serv_allocate':
mkl_memory.c:(.text+0x19eb): undefined reference to `dlopen'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x1a08): undefined reference to `dlsym'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x1a49): undefined reference to `dlclose'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x1a5c): undefined reference to `dlsym'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x1a72): undefined reference to `dlsym'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x1b60): undefined reference to `dlerror'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x1bca): undefined reference to `dlopen'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x1c6f): undefined reference to `dlopen'
/home/linuxbrew/.linuxbrew/bin/ld: /opt/ibm/ILOG/CPLEX_Studio221/cplex/lib/x86-64_linux/static_pic/libcplex.a(mkl_memory_patched.o): in function `mkl_serv_malloc':
mkl_memory.c:(.text+0x227f): undefined reference to `dlopen'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x229c): undefined reference to `dlsym'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x22dd): undefined reference to `dlclose'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x22f0): undefined reference to `dlsym'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x2306): undefined reference to `dlsym'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x23f4): undefined reference to `dlerror'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x245e): undefined reference to `dlopen'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x2502): undefined reference to `dlopen'
/home/linuxbrew/.linuxbrew/bin/ld: /opt/ibm/ILOG/CPLEX_Studio221/cplex/lib/x86-64_linux/static_pic/libcplex.a(mkl_memory_patched.o): in function `mkl_serv_deallocate':
mkl_memory.c:(.text+0x3b0e): undefined reference to `dlopen'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x3b35): undefined reference to `dlsym'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x3b76): undefined reference to `dlclose'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x3b89): undefined reference to `dlsym'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x3b9f): undefined reference to `dlsym'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x3c8b): undefined reference to `dlerror'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x3cfa): undefined reference to `dlopen'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x3da0): undefined reference to `dlopen'
/home/linuxbrew/.linuxbrew/bin/ld: /opt/ibm/ILOG/CPLEX_Studio221/cplex/lib/x86-64_linux/static_pic/libcplex.a(mkl_memory_patched.o): in function `mkl_serv_free':
mkl_memory.c:(.text+0x42cb): undefined reference to `dlopen'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x42e8): undefined reference to `dlsym'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x4329): undefined reference to `dlclose'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x433c): undefined reference to `dlsym'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x4352): undefined reference to `dlsym'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x4440): undefined reference to `dlerror'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x44aa): undefined reference to `dlopen'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x455e): undefined reference to `dlopen'
/home/linuxbrew/.linuxbrew/bin/ld: /opt/ibm/ILOG/CPLEX_Studio221/cplex/lib/x86-64_linux/static_pic/libcplex.a(mkl_memory_patched.o): in function `mkl_serv_free_buffers':
mkl_memory.c:(.text+0x58c3): undefined reference to `dlopen'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x58ea): undefined reference to `dlsym'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x592b): undefined reference to `dlclose'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x593e): undefined reference to `dlsym'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x5954): undefined reference to `dlsym'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x5a41): undefined reference to `dlerror'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x5ab0): undefined reference to `dlopen'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x5b57): undefined reference to `dlopen'
/home/linuxbrew/.linuxbrew/bin/ld: /opt/ibm/ILOG/CPLEX_Studio221/cplex/lib/x86-64_linux/static_pic/libcplex.a(mkl_memory_patched.o): in function `mkl_serv_thread_free_buffers':
mkl_memory.c:(.text+0x6096): undefined reference to `dlopen'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x60b3): undefined reference to `dlsym'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x60f4): undefined reference to `dlclose'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x6107): undefined reference to `dlsym'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x611d): undefined reference to `dlsym'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x620b): undefined reference to `dlerror'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x6275): undefined reference to `dlopen'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x6319): undefined reference to `dlopen'
/home/linuxbrew/.linuxbrew/bin/ld: /opt/ibm/ILOG/CPLEX_Studio221/cplex/lib/x86-64_linux/static_pic/libcplex.a(mkl_memory_patched.o): in function `mkl_serv_realloc':
mkl_memory.c:(.text+0x734e): undefined reference to `dlopen'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x736b): undefined reference to `dlsym'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x73ac): undefined reference to `dlclose'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x73bf): undefined reference to `dlsym'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x73d5): undefined reference to `dlsym'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x74c3): undefined reference to `dlerror'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x752d): undefined reference to `dlopen'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x75d1): undefined reference to `dlopen'
/home/linuxbrew/.linuxbrew/bin/ld: /opt/ibm/ILOG/CPLEX_Studio221/cplex/lib/x86-64_linux/static_pic/libcplex.a(mkl_memory_patched.o): in function `mkl_serv_jit_alloc':
mkl_memory.c:(.text+0x7b57): undefined reference to `dlopen'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x7b74): undefined reference to `dlsym'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x7bb5): undefined reference to `dlclose'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x7bc8): undefined reference to `dlsym'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x7bde): undefined reference to `dlsym'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x7ccc): undefined reference to `dlerror'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x7d36): undefined reference to `dlopen'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x7dea): undefined reference to `dlopen'
/home/linuxbrew/.linuxbrew/bin/ld: /opt/ibm/ILOG/CPLEX_Studio221/cplex/lib/x86-64_linux/static_pic/libcplex.a(mkl_memory_patched.o): in function `mkl_serv_jit_free':
mkl_memory.c:(.text+0x8d25): undefined reference to `dlopen'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x8d42): undefined reference to `dlsym'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x8d83): undefined reference to `dlclose'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x8d96): undefined reference to `dlsym'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x8dac): undefined reference to `dlsym'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x8e9a): undefined reference to `dlerror'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x8f04): undefined reference to `dlopen'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x8fa8): undefined reference to `dlopen'
/home/linuxbrew/.linuxbrew/bin/ld: /opt/ibm/ILOG/CPLEX_Studio221/cplex/lib/x86-64_linux/static_pic/libcplex.a(mkl_memory_patched.o): in function `mm_cleanup_thread_cbk':
mkl_memory.c:(.text+0x965d): undefined reference to `dlopen'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x967a): undefined reference to `dlsym'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x96bb): undefined reference to `dlclose'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x96ce): undefined reference to `dlsym'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x96e4): undefined reference to `dlsym'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x97d2): undefined reference to `dlerror'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x983c): undefined reference to `dlopen'
/home/linuxbrew/.linuxbrew/bin/ld: mkl_memory.c:(.text+0x98e0): undefined reference to `dlopen'
/home/linuxbrew/.linuxbrew/bin/ld: /opt/ibm/ILOG/CPLEX_Studio221/cplex/lib/x86-64_linux/static_pic/libcplex.a(mkl_memory_patched.o): in function `_Init_MKL_Loader':
mkl_memory.c:(.text+0x9a30): undefined reference to `dladdr'
/home/linuxbrew/.linuxbrew/bin/ld: /opt/ibm/ILOG/CPLEX_Studio221/cplex/lib/x86-64_linux/static_pic/libcplex.a(libc_is_static_interface.o): in function `mkl_serv_libc_is_static':
libc_is_static_interface.c:(.text+0x10): undefined reference to `dladdr'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
18:21:44: The process "/home/juanjqo/Qt/Tools/CMake/bin/cmake" exited with code 1.
Error while building/deploying project cplex_interface_test (kit: Desktop Qt 6.3.1 GCC 64bit)
When executing step "Build"
```